web/packages/teleport/src/Login/useLogin.ts

```diff
@@ -69,8 +70,25 @@ export default function useLogin() {
 
   useEffect(() => {
     if (session.isValid()) {
-      history.replace(cfg.routes.root);
-      return;
+      try {
+        const redirectUrlWithBase = new URL(getEntryRoute());
+        const matched = matchPath(redirectUrlWithBase.pathname, {
+          path: cfg.routes.samlIdpSso,
+          strict: true,
+          exact: true,
+        });
+        if (matched) {
+          history.push(redirectUrlWithBase, true);
```

Comment:
should it be push or replace?

Reply:
`push` so back navigation works if user wishes to. I checked other existing login redirector uses push with page refresh too.